test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/Rendering/HtmlHelperValidationSummaryTest.cs

```diff
@@ -109,15 +109,15 @@ public class HtmlHelperValidationSummaryTest
                     "<li>HtmlEncode[[This is an error for Property3.]]</li>" + Environment.NewLine +
                     "</ul></div>";
                 var divWithAllErrors = "<div class=\"HtmlEncode[[validation-summary-errors]]\" data-valmsg-summary=\"HtmlEncode[[true]]\"><ul>" +
-                    "<li>HtmlEncode[[This is an error for Property3.Property2.]]</li>" + Environment.NewLine +
-                    "<li>HtmlEncode[[This is an error for Property3.OrderedProperty3.]]</li>" + Environment.NewLine +
-                    "<li>HtmlEncode[[This is an error for Property3.OrderedProperty2.]]</li>" + Environment.NewLine +
-                    "<li>HtmlEncode[[This is an error for Property3.]]</li>" + Environment.NewLine +
+                    "<li>HtmlEncode[[This is an error for the model root.]]</li>" + Environment.NewLine +
+                    "<li>HtmlEncode[[This is another error for the model root.]]</li>" + Environment.NewLine +
                     "<li>HtmlEncode[[This is an error for Property2.]]</li>" + Environment.NewLine +
                     "<li>HtmlEncode[[This is another error for Property2.]]</li>" + Environment.NewLine +
                     "<li>HtmlEncode[[The value '' is not valid for Property2.]]</li>" + Environment.NewLine +
-                    "<li>HtmlEncode[[This is an error for the model root.]]</li>" + Environment.NewLine +
-                    "<li>HtmlEncode[[This is another error for the model root.]]</li>" + Environment.NewLine +
+                    "<li>HtmlEncode[[This is an error for Property3.]]</li>" + Environment.NewLine +
+                    "<li>HtmlEncode[[This is an error for Property3.OrderedProperty2.]]</li>" + Environment.NewLine +
+                    "<li>HtmlEncode[[This is an error for Property3.OrderedProperty3.]]</li>" + Environment.NewLine +
+                    "<li>HtmlEncode[[This is an error for Property3.Property2.]]</li>" + Environment.NewLine +
```

Comment:
> Do not depart from the order entries are added to the ModelStateDictionary.

Shouldn't the caller `ValidationHelper` in this case - re-sort things in the order it expects rather than expect the dictionary to preserve order? Preserving insertion order isn't a behavior guaranteed by dictionaries in general.